File: CSharp/PetShop/Source/PetShop.UI/Controls/ProductsControl.ascx.cs
using System;
using System.Web.UI;
using System.Web.UI.WebControls;

using PetShop.Business;

namespace PetShop.UI.Controls
{
    public partial class ProductsControl : UserControl
    {
        /// <summary>
        /// Rebind control 
        /// </summary>
        protected void PageChanged(object sender, DataGridPageChangedEventArgs e)
        {
            //reset index
            productsList.CurrentPageIndex = e.NewPageIndex;

            //get category id
            string categoryId = Request.QueryString["categoryId"];

            //bind data
            productsList.DataSource = Category.GetByCategoryId(categoryId).Products;
            productsList.DataBind();
        }
    }
}

File: tags/CSLA-3.0.1-Samples/VisualBasic/PetShop/Source/PetShop.UI/Controls/ProductsControl.ascx.vb
Imports PetShop.Business

Partial Public Class ProductsControl
    Inherits System.Web.UI.UserControl

    ''' <summary>
    ''' Rebind control
    ''' </summary>
    Protected Sub PageChanged(ByVal sender As Object, ByVal e As DataGridPageChangedEventArgs)
        'reset index
        productsList.CurrentPageIndex = e.NewPageIndex

        'get category id
        Dim categoryId As String = Request.QueryString("categoryId")

        'bind data
        productsList.DataSource = Category.GetByCategoryId(categoryId).Products
        productsList.DataBind()
    End Sub

End Class